### PR TITLE
Adds feral hogs, maint slugs, and replaces the HOS and cargo's pets

### PR DESCRIPTION
**Pull request by Spyroshark**
## About The Pull Request

Adds Maintenance Slugs and Feral Hogs to the game. Slugs are passive and can be found in maint on occasion, and leave a trail of slippery slime as they crawl around. Their "udders" (aka "slime glands" or "sludders" if you're so inclined) can be "milked" to collect their slime, which is slippery, cleans tiles, and works as a surgical chem.

Feral Hogs are a neutral mob that will turn hostile when anything gets too close. They currently only spawn during the Animal Infestation event, and should be a very rare sight. I will consider adding more ways to find hogs later on. (hog antag 2024). also they're insanely strong so don't spawn 30-40 of them please

This PR also removes the HOS' pet spider and cargo's sloth in favor of two new pets, Lieutenant Hoggison and Glubby. Hoggison is a security pig that gets his own pen by the security lockers and is much more passive then normal hogs. Glubby is a unique slug that replaces Paperwork the sloth.

## Why It's Good For The Game

I've never seen anyone care about the sloth, and the hos' spider is a BAT. I think these are more fun replacements. Slugs in maint are fun and uncommon enough to not get too annoying, and the rare occasional hog attack is incredibly funny to me.

## Changelog

:cl:
add: Maint Slugs! They leave behind a slippery trail and can be "milked" if so desired. Cargo starts with a unique one.
add: Feral Hogs! Technically a neutral animal, they become aggressive if you get too close. Security has a private hog.
del: Sgt. Araneus and Paperwork have been sent off to kindly old man's space-farm, where they will happily live out the rest of their natural lives.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/retaliate/hog.dm [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	var/territorial = TRUE
+	var/datum/action/cooldown/mob_cooldown/charge/basic_charge/charge
+
+/mob/living/simple_animal/hostile/retaliate/hog/Initialize(mapload)
+	. = ..()
+	charge = new /datum/action/cooldown/mob_cooldown/charge/basic_charge()
+	charge.Grant(src)
+	if(territorial)
+		AddComponent(/datum/component/connect_range, src, list(COMSIG_ATOM_ENTERED = PROC_REF(checkEntered)), 1, FALSE)
+	if(name == "feral hog") //don't rename renamed hogs
```
Comment (Kapu1178) [inline code/modules/mob/living/simple_animal/hostile/retaliate/hog.dm:37]:
```suggestion
	if(name == initial(name)) //don't rename renamed hogs
```

**Review by Kapu1178 — CHANGES_REQUESTED** [review]
initial

File: code/modules/mob/living/simple_animal/hostile/retaliate/hog.dm [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	charge.Trigger(target = target)
+
+/mob/living/simple_animal/hostile/retaliate/hog/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	if(prob(15))
+		playsound(src, 'sound/creatures/hog/hoggrunt.ogg', 50, TRUE)
+
+/mob/living/simple_animal/hostile/retaliate/hog/proc/hogAlert() //YOU HAVE ALERTED THE HOG
+	var/image/alert_overlay = image('icons/mob/animal.dmi', src, "hog_alert_overlay", layer+1)
+	flick_overlay_view(alert_overlay, src, 1.5 SECONDS)
```
Comment (Kapu1178) [inline code/modules/mob/living/simple_animal/hostile/retaliate/hog.dm:71]: Use vis_contents for this.

```
var/obj/effect/overlay/vis/overlay = new()
overlay .icon = 'icons/mob/animal.dmi'
overlay .icon_state = 'icons/mob/animal.dmi
vis_contents += overlay
QDEL_IN(overlay, 1.5 SECONDS)
```

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/retaliate/hog.dm (hunk 2 above)
Comment (Spyroshark) [reply code/modules/mob/living/simple_animal/hostile/retaliate/hog.dm]: done

**Comment by Cenrus** [comment]
Add the ridable element to the hog, cows use it too

**Comment by Spyroshark** [comment]
> Add the ridable element to the hog, cows use it too

i was told specifically to not add ridable hogs. however perhaps it is better to die standing then to live kneeling

File: code/modules/mob/living/simple_animal/hostile/retaliate/hog.dm [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+/mob/living/simple_animal/hostile/retaliate/hog/proc/checkEntered(datum/source, atom/movable/arrived)
+	if(arrived == src)
+		return
+	if(!ismob(arrived))
+		return
+	Retaliate()
+
+/mob/living/simple_animal/hostile/retaliate/hog/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	if(prob(15))
```
Comment (FunnyLegJiggle) [inline code/modules/mob/living/simple_animal/hostile/retaliate/hog.dm:54]: does not check for the alive state of the mob, allowing HOG grunting from the afterlife if the body is moved

**Comment by Spyroshark** [comment]
oops

**Pull request closed without merging**